bin_sort/BinSort.java

```diff
@@ -0,0 +1,75 @@
+import java.util.Random;
+
+class Node {
+	public double data;
+	public Node next;
+
+	public Node(double data) {
+		this.data = data;
+		next = null;
+	}
+}
+
+// Time Complexity O(n)-> avg case and O(n^2)-> Worst Case
+// constraints:- 0<=A[i]<1
+public class BinSort {
+	public static void binSort(Node[] b, double[] a) {
+		for (int i = 0; i < a.length; i++) {
+			int ins = (int) (a[i] * 10);

```

Comment:
@Hsankesara https://www.tutorialspoint.com/java/number_floor.htm Please refer this once.
If you have any link stating floating point please share.